cmd/podman/buildfarm/prune.go

```diff
@@ -0,0 +1,40 @@
+package buildfarm
+
+import (
+	"github.com/containers/podman/v4/cmd/podman/common"
+	"github.com/containers/podman/v4/cmd/podman/registry"
+	"github.com/spf13/cobra"
+)
+
+var (
+	buildfarmPruneDescription = `Prune untagged and optionally unused images from a farm`
+
+	pruneCommand = &cobra.Command{
+		Use:               "prune [options] [FARM...]",
+		Short:             "Prune untagged and optionally unused images on the nodes in a farm",
+		Long:              buildfarmPruneDescription,
+		RunE:              prune,
+		ValidArgsFunction: common.AutoCompleteFarms,
+		Example: `podman buildfarm prune myfarm
+  podman buildfarm prune --all`,
+	}
+
+	// Temporary struct to hold cli values.
+	pruneOpts = struct {
+		All bool
+	}{}
+)
+
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Command: pruneCommand,
+		Parent:  buildfarmCmd,
+	})
+	flags := pruneCommand.Flags()
+	flags.BoolVarP(&pruneOpts.All, "all", "a", false, "Prune images from all farms")
```

Comment:
`--all` already has a meaning for other prune commands to remove non-dangling but unused images

Reply:
Right, I thought it would be a good interface to have for users if they want to prune all the images on all the machines in all their farms instead of having to pass in all the farm names to the command. 
I am open to removing the `--all` here if it can be confusing for users but I think the help menu and docs should be clear enough of the purpose of how `--all` is different here.

Reply:
As mentioned elsewhere, I think we should drop as much as possible from this PR to make it less contentious.  At the moment, it remains unclear how `prune` should behave in details, so I think it's easier to drop (or ignore) it for now.